packages/task/src/node/task-abstract-line-matcher.ts

```diff
@@ -295,7 +295,7 @@ export abstract class AbstractLineMatcher {
             const regexp = new RegExp(this.activePattern.regexp);
             const regexMatches = regexp.exec(line);
             if (regexMatches) {
-                if (this.activePattern.kind !== undefined && this.cachedProblemData.kind !== undefined) {
+                if (this.activePattern.kind !== undefined && this.cachedProblemData.kind === undefined) {
                     this.cachedProblemData.kind = this.activePattern.kind;
                 }
```

Comment:
This could be simplified to
```suggestion
                this.cachedProblemData.kind ??= this.activePattern.kind;
```